### PR TITLE
[hadd] properly change compression when levels are the same

### LINKED ISSUE #11245: hadd doesn't recompress for different algorithm at same level
- [x] Checked for duplicates

### Describe the bug

When changing the compression settings using hadd, using the same compression level used by the input file won't trigger a recompression of the branches in the output file, even with a different algorithm.

ie. Passing -f301 with a 101 compressed input file results in a "301" compressed file but with 101 compressed branches.

### Expected behavior

I would expect the output file's branches to be recompressed with the provided compression algorithm.

ie. Passing -f301 with a 101 compressed input file results in a genuine 301 compressed file with 301 compressed branches.

### To Reproduce

1. Get the compression settings of any .root file, for example the 101 compressed [dimuon dataset](http://opendata.web.cern.ch/record/12341).
2. Run `hadd` and change the compression to a different algorithm of the same level, for example 101 -> 201 or 403 -> 303:
`hadd -f201 fake_dimuon_201.root orig_dimuon_101.root`
3. You can validate the weird behaviour by performing an intermediate conversion:
`hadd -f202 real_202_out.root orig_dimuon_101.root` -> `hadd -f201 real_dimuon_201.root real_dimuon_202.root`

### Setup

1. ROOT Version 6.27/01
2. Linux CentOS 8 + Linux Debian 11
3. Built from source

### PR DESCRIPTION
**Pull request by silverweed**
Currently hadd is only checking the compression levels to determine whether fast merging should be used or not. This is wrong, as it will incorrectly skip compression change when a file is changed from e.g. 101 to 201. With this commit, hadd (actually, TFileMerger) will check for equality of compression settings instead.

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #11245

### CODE REVIEW

File: main/src/hadd.cxx [hunk 1]
```diff
@@ -483,7 +483,7 @@ int main( int argc, char **argv )
       } else {
          if (!keepCompressionAsIs && merger.HasCompressionChange()) {
             // Don't warn if the user any request re-optimization.
```
Comment (vepadulano) [inline main/src/hadd.cxx:485]: Maybe not for this PR, but this comment has surely at least a typo, possibly the comment is misplaced altogether. @pcanal what do you think?

**Review by vepadulano — CHANGES_REQUESTED** [review]
The change makes sense to me but we surely need a test in roottest for this

File: main/src/hadd.cxx (hunk 1 above)
Comment (silverweed) [reply main/src/hadd.cxx:485]: It's a typo, it means that the warning is not given if the user explicitly asked for reoptimization (at least, that's how I interpret it). I'll fix the typo, thanks.

**Review by jblomer — APPROVED** [review]

**Comment by jblomer** [comment]
As this is a bug fix, we should add it to the release notes (or open an issue for it so that we don't forget to add it later).

**Comment by silverweed** [comment]
@jblomer I put it in the "fixed in 6.34" project: is that enough?

**Comment by silverweed** [comment]
@vepadulano created roottest for this case https://github.com/root-project/roottest/pull/1150

**Review by dpiparo — APPROVED** [review]

**Review by vepadulano — APPROVED** [review]
Thank you!

**Comment by pcanal** [comment]
> > As this is a bug fix, we should add it to the release notes (or open an issue for it so that we don't forget to add it later).
>  I put it in the "fixed in 6.34" project: is that enough?

@dpiparo Is it actually enough or do we also need to open an issue? (I though it was the later)

**Comment by vepadulano** [comment]
Yes I believe we need to open an issue and marked it fixed

**Pull request merged**